eth/peer.go

```diff
@@ -278,7 +280,26 @@ func (p *peer) Handshake(td *big.Int, head common.Hash, genesis common.Hash) err
 			GenesisBlock:    genesis,
 		})
 	}()
-	// In the mean time retrieve the remote status message
+	go func() {
+		errc <- p.readStatus(&status, genesis)
+	}()
+	timeout := time.NewTimer(handshakeTimeout)
+	defer timeout.Stop()
+	for i := 0; i < 2; i++ {
```

Comment:
Since `i` is not used, this can be done as a func...hmm...but then the `return` statements have to be forwarded.